snakegame.py

```diff
@@ -94,23 +98,43 @@ def rungame(self, init_length, init_speed, fruits_no):
 		print("Runing new game.")
 
 		# Game init
-		snake = Snake(3, 500, 500)
+		snake = Snake()
 		snake.joints = [Joint(5,8), Joint(4,8), Joint(3,8)]
 		fruits = []
 		for _ in range(fruits_no):
-			fruits.append(Joint(random.randrange(16), random.randrange(16)))
+			fruits.append(Joint(random.randrange(self.fieldSizeX), random.randrange(self.fieldSizeY)))
+		walls = []
+		for i in range(self.fieldSizeX):
+			walls.append(Joint(i, 0))
+			walls.append(Joint(i, self.fieldSizeY - 1))
+		for i in range(self.fieldSizeY - 2):
+			walls.append(Joint(0, i + 1))
+			walls.append(Joint(self.fieldSizeX - 1, i + 1))
 		speed = init_speed
 
+		period = 1 / speed
+		
+		time_stamp = time.monotonic()
+
 		while self.running:
-			period = 1 / speed
-			time.sleep(period)
-
-			key = self.getPlayerLastInput()
-			if key != None:
-				snake.applyPlayerInput(key)
-				
-			snake.updateSnakePosition()
-			self.running = self.screen.drawScreen(snake, fruits)
+
+			if time.monotonic() - time_stamp > period:
+					
+				time_stamp = time.monotonic()
+
+				key = self.getPlayerLastInput()
+				if key != None:
+					snake.applyPlayerInput(key)
+					
+				snake.updateSnakePosition()
+
+				if snake.isInCollisionWith(snake.joints[1:]) or snake.isInCollisionWith(walls):
```

Comment:
isInCollisionWith(walls) might be optimized. You don't need to check every single wall joint. The better way is to check if head of snake is between x and y play field borders. This would reduce maximal computing time since this implementation is O(n+m) where n and m are dimensions of play field.

Reply:
Goal of wall functionality is really to check whether the snake did not collide with one of walls. These might appear also in the middle of gamefield. And then such optimization would fail. In fact, `borderCollision` functionality is not implemented. It is 'hacked' with walls laid around as border. `borderCollision` will be implemented according to your suggestion.